base/applications/cmdutils/tasklist/tasklist.c

```diff
@@ -0,0 +1,398 @@
+/*
+ * PROJECT:     ReactOS Tasklist Command
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Displays a list of currently running processes on the computer.
+ * COPYRIGHT:   Copyright 2020 He Yang (1160386205@qq.com)
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#define WIN32_NO_STATUS
+#include <windows.h>
+#include <ndk/psfuncs.h>
+
+#include <conutils.h>
+
+#include "resource.h"
+
+
+#define NT_SYSTEM_QUERY_MAX_RETRY 5
+
+
+#define COLUMNWIDTH_IMAGENAME     25
+#define COLUMNWIDTH_PID           8
+#define COLUMNWIDTH_SESSION       11
+#define COLUMNWIDTH_MEMUSAGE      12
+
+
+#define HEADER_STR_MAXLEN 64
+
+
+static WCHAR opHelp[] = L"?";
+static WCHAR opVerbose[] = L"v";
+static PWCHAR opList[] = {opHelp, opVerbose};
+
+
+#define OP_PARAM_INVALID -1
+
+#define OP_PARAM_HELP 0
+#define OP_PARAM_VERBOSE 1
+
+
+typedef NTSTATUS (NTAPI *NT_QUERY_SYSTEM_INFORMATION) (
+    SYSTEM_INFORMATION_CLASS SystemInformationClass,
+    PVOID SystemInformation,
+    ULONG InformationLength,
+    PULONG ResultLength);
+
+VOID PrintSplitLine(int Length)
+{
+    for (int i = 0; i < Length; i++)
+    {
+        ConPrintf(StdOut, L"=");
+    }
+}
+
+VOID PrintHeader()
+{
+    WCHAR lpstrImageName[HEADER_STR_MAXLEN];
+    WCHAR lpstrPID[HEADER_STR_MAXLEN];
+    WCHAR lpstrSession[HEADER_STR_MAXLEN];
+    WCHAR lpstrMemUsage[HEADER_STR_MAXLEN];
+
+    LoadStringW(GetModuleHandle(NULL), IDS_HEADER_IMAGENAME, lpstrImageName, HEADER_STR_MAXLEN);
+    LoadStringW(GetModuleHandle(NULL), IDS_HEADER_PID, lpstrPID, HEADER_STR_MAXLEN);
+    LoadStringW(GetModuleHandle(NULL), IDS_HEADER_SESSION, lpstrSession, HEADER_STR_MAXLEN);
+    LoadStringW(GetModuleHandle(NULL), IDS_HEADER_MEMUSAGE, lpstrMemUsage, HEADER_STR_MAXLEN);
+
+    ConPrintf(
+        StdOut, L"%-*.*ls %*.*ls %*.*ls %*.*ls",
+        COLUMNWIDTH_IMAGENAME, COLUMNWIDTH_IMAGENAME, lpstrImageName,
+        COLUMNWIDTH_PID, COLUMNWIDTH_PID, lpstrPID,
+        COLUMNWIDTH_SESSION, COLUMNWIDTH_SESSION, lpstrSession,
+        COLUMNWIDTH_MEMUSAGE, COLUMNWIDTH_MEMUSAGE, lpstrMemUsage);
+
+    ConPrintf(StdOut, L"\n");
+
+    PrintSplitLine(COLUMNWIDTH_IMAGENAME);
+    ConPrintf(StdOut, L" ");
+    PrintSplitLine(COLUMNWIDTH_PID);
+    ConPrintf(StdOut, L" ");
+    PrintSplitLine(COLUMNWIDTH_SESSION);
+    ConPrintf(StdOut, L" ");
+    PrintSplitLine(COLUMNWIDTH_MEMUSAGE);
+
+    ConPrintf(StdOut, L"\n");
+}
+
+// Print spaces
+BOOL PrintSpace(int SpaceNum)
+{
+    for (; SpaceNum; SpaceNum--)
+    {
+        ConPrintf(StdOut, L" ");
+    }
+    return TRUE;
+}
+
+// Print a string, aligned to left.
+// MaxWidth is the width for printing.
+INT PrintString(LPCWSTR String, int MaxWidth)
+{
+    return ConPrintf(StdOut, L"%-*.*ls", MaxWidth, MaxWidth, String);
+}
+
+// Print a string, aligned to right.
+// MaxWidth is the width for printing.
+// return FALSE if number length is longer than MaxWidth
+BOOL PrintNum(long long Number, int MaxWidth)
```

Comment:
Please use NT types (ULONGLONG, INT etc...) or things like INT32, INT64 ...

Reply:
ok!  I will fix it tomorrow